cpp/BUILD.bazel

```diff
@@ -83,10 +83,35 @@ genrule(
         cp -f $(locations libray_api.so) "$$WORK_DIR/python/ray/core/src/ray/cpp/" &&
         TEMP_DIR="$$WORK_DIR/cpp/example" &&
         rm -rf $$TEMP_DIR/thirdparty &&
-        mkdir -p "$$TEMP_DIR/thirdparty/include/" &&
+        BOOST_DIR="$$TEMP_DIR/thirdparty/include/boost/" &&
+        mkdir -p "$$BOOST_DIR" &&
         mkdir -p "$$TEMP_DIR/thirdparty/lib/" &&
         cp -f -r $$WORK_DIR/external/msgpack/include/* "$$TEMP_DIR/thirdparty/include" &&
-        cp -f -r "$$WORK_DIR/external/boost/boost/" "$$TEMP_DIR/thirdparty/include" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/archive" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/bind" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/callable_traits" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/concept" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/config" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/container" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/container_hash" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/core" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/detail" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/dll" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/exception" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/filesystem" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/functional" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/io" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/iterator" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/lexical_cast" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/move" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/mpl" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/optional" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/parameter" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/preprocessor" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/system" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/type_traits" "$$BOOST_DIR" &&
+        cp -f -r "$$WORK_DIR/external/boost/boost/utility" "$$BOOST_DIR" &&
+        cp -f -r $$WORK_DIR/external/boost/boost/*.hpp "$$BOOST_DIR" &&
```

Comment:
All these libraries are required? Still a lot to me.

Reply:
Yes, some sub library reference other libraries, and these libraries are from [boost.bcp](https://www.boost.org/doc/libs/1_71_0/tools/bcp/doc/html/index.html) analyzation.

Reply:
Before this PR, the size of boost headers is 161M, after this PR, the size will be 20M.